distribution/xfer/transfer.go

```diff
@@ -19,6 +21,11 @@ func (e DoNotRetry) Error() string {
 	return e.Err.Error()
 }
 
+func IsDoNotRetryError(err error) bool {
```

Comment:
The linter may fail on this as it's exported, but doesn't have  a `godoc`;  perhaps something like:

```go
// IsDoNotRetryError returns true if the error is caused by DoNotRetry error,
// and the transfer should not be retried.
```